lib/modules/dosomething/dosomething_user/dosomething_user.module

```diff
@@ -234,12 +234,15 @@ function dosomething_user_form_alter(&$form, $form_state, $form_id) {
     break;
 
     case 'user_profile_form':
+      $form['#validate'][] = 'dosomething_user_validate_password_length';
     case 'user_register_form':
       // Gather helper text.
       _dosomething_user_register_helper_text($form);
       // Conditionally displays certain fields based on configuration settings.
       _dosomething_user_global_display_fields($form);
 
+      $form['#validate'][] = 'dosomething_user_validate_password_length';
```

Comment:
instead of adding this in two places, it should work the way it was where both 
case 'user_profile_form' & case 'user_register_form' use the same code below